syz-verifier/main.go

```diff
@@ -267,6 +273,76 @@ func (srv *RPCServer) NextExchange(a *rpctype.NextExchangeArgs, r *rpctype.NextE
 	return nil
 }
 
+// newResult is called when a Runner sends a new Result. It returns true if all
+// Results from the corresponding programs have been received and they can be
+// sent for verification. Otherwise, it returns false.
+func (srv *RPCServer) newResult(res *verf.Result, idx int) bool {
+	prog := srv.progs[idx]
+	if prog.idx == idx {
```

Comment:
Can this condition happen? How? If yes, do we want to silently ignore it?

Reply:
`prog.idx` will always be equal to `idx` since the `srv.progs` map is indexed by the program's indices